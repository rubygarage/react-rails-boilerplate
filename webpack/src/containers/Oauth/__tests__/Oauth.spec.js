import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'
import OauthContainer, { Oauth } from 'containers/auth/Oauth'

describe('<Oauth />', () => {
  const defaultProps = {
    color: 'blue-dark',
    oauthRequest: jest.fn(),
    provider: 'facebook',
    type: 'icon-fb'
  }

  it('handleOauthRequest()', () => {
    const component = shallow(<Oauth {...defaultProps} />)
    const event = { preventDefault: jest.fn() }
    component.instance().handleOauthRequest(event)

    expect(defaultProps.oauthRequest).toHaveBeenCalledWith(defaultProps.provider)
  })

  it('renders oauth link', () => {
    const component = shallow(<Oauth {...defaultProps} />)

    expect(component).toMatchSnapshot()
  })

  it('renders oauth button', () => {
    const props = {
      ...defaultProps,
      button: true
    }
    const component = shallow(<Oauth {...props} />)

    expect(component).toMatchSnapshot()
  })

  it('maps dispatch to props', () => {
    const store = configureStore()({})
    const container = shallow(<OauthContainer store={store} {...defaultProps} />)

    expect(container.props()).toEqual(expect.objectContaining({
      oauthRequest: expect.any(Function)
    }))
  })
})
