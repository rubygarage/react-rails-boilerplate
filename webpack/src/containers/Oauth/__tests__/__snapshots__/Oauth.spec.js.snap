// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Oauth /> renders oauth button 1`] = `
<OauthButton
  color="blue-dark"
  onClick={[Function]}
  type="icon-fb"
/>
`;

exports[`<Oauth /> renders oauth link 1`] = `
<OauthLink
  onClick={[Function]}
  type="icon-fb"
/>
`;
